src/mmap.rs

```diff
@@ -181,8 +182,21 @@ impl GuestRegionMmap {
         }
     }
 
-    fn as_volatile_slice(&self) -> VolatileSlice {
-        self.mapping.as_volatile_slice()
+    /// Convert an absolute address into an address space (GuestMemory)
+    /// to a host pointer, or return None if it is out of bounds.
+    pub fn get_host_address(&self, addr: MemoryRegionAddress) -> Option<*mut u8> {
+        // Not sure why wrapping_offset is not unsafe.  Anyway this
+        // is safe because we've just range-checked addr using check_address.
+        self.check_address(addr)
+            .map(|addr| self.as_ptr().wrapping_offset(addr.raw_value() as isize))
```

Comment:
Hard to say why one is unsafe while the other isn't; they compile down to different intrinsics whose sources I couldn't find during a cursory search :( 

Anyway, apologies for not providing a more complete comment the first time: this still looks potentially wrong because of the `as isize` conversion. For example, in the 32 bit case, the reinterpretation `raw_value() as isize` might turn intro a negative number if the memory region and `addr` are large enough, and the result will be wrong (for 64 bit it's still theoretically possible, but kinda unrealistic). I guess we can just check against this condition as well.

Reply:
The important thing is that the `MemoryRegionAddress`'s u64 has been checked already to be in range of `self.len()`, so it must fit in 32 bits.

At this point, even if it's a negative number after the `isize` cast, it wraps around just fine.

Reply:
Oh yeah, I forgot how these numbers actually work :( My bad